test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php

```diff
@@ -217,12 +217,31 @@ public function testReturnsNullWithInternalFunctions() : void
 
     public function testCanAutoloadPsr4ClassesInPotentiallyMultipleDirectories() : void
     {
+        spl_autoload_register([$this, 'autoload']);
+
         self::assertNotNull(
             (new AutoloadSourceLocator($this->astLocator))
                 ->locateIdentifier(
                     $this->getMockReflector(),
-                    new Identifier(ExampleFinder::class, new IdentifierType(IdentifierType::IDENTIFIER_CLASS))
+                    new Identifier(AutoloadableClassWithTwoDirectories::class, new IdentifierType(IdentifierType::IDENTIFIER_CLASS))
                 )
         );
+
+        spl_autoload_unregister([$this, 'autoload']);
```

Comment:
Please also ensure that the class does not exist at this point in the execution.